File: runtime/ruby/units/table/_class.rb
# One slot per element: a `[type_symbol, callable]` pair for funcref tables, or `nil` for a null slot. call_indirect compares type keys, not module-local indices, so a shared table stays consistent across modules.
def initialize(size, max = nil)
  @slots = Array.new(size)
  @max = max
end

def wasm_kind = :table

def size = @slots.size

File: runtime/python/units/table/_class.py
# One slot per element: a [type_key, callable] pair for funcref tables, or
# None for a null slot. call_indirect compares type keys, not module-local indices, so a shared table stays consistent across modules.
wasm_kind = "table"

def __init__(self, size, max=None):
    self._slots = [None] * size
    self._max = max

def size(self):
    return len(self._slots)
